### PR TITLE
feat(validity): add validation for min and max values #242

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #242

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

File: src/base-input.js [hunk 1]
```diff
@@ -692,27 +708,27 @@ export default class BaseInput extends LitElement {
        */
       if ((!this.value || this.value.length === 0) && this.required) {
         this.validity = 'valueMissing';
-        this.setCustomValidity = this.setCustomValidityValueMissing;
+        this.setCustomValidity = this.setCustomValidityValueMissing || '';
```
Comment (jordanjones243) [inline src/base-input.js:711]: Not a comment for review, but what does adding the `| | ''` do? I see it all throughout the code when you are setting `this.setCustomValidity`.

File: src/base-input.js [hunk 2]
```diff
@@ -734,14 +750,31 @@ export default class BaseInput extends LitElement {
...
+   * @returns {void}
+   */
+  configureDataForType() {
+    if (this.type === 'month-day-year') {
+      this.dateStrLength = 10;
+    } else if (this.type === 'month-year') {
+      this.dateStrLength = 5;
+    } else if (this.type === 'month-fullYear') {
+      this.dateStrLength = 7;
+    } else if (this.type === 'year-month-day') {
```
Comment (jordanjones243) [inline src/base-input.js:776]: Add this to the first `if` statement: `if (this.type === ('month-day-year' || 'year-month-day')) {

File: demo/demo.md [hunk 3]
```diff
@@ -140,6 +140,90 @@ Note: Setting the `value` to `undefined` will also reset the element.
...
+  <span slot="trigger">See code</span>
+
+  ```html
+  <auro-input id="demoMinNumber" type="numeric" min="10" setCustomValidityRangeUnderflow="The selected value is below the defined minimum.">
+    <span slot="label">Choose a number</span>
+  </auro-input>
+  ```
+</auro-accordion>
+
 ### Pattern
```
Comment (jordanjones243) [inline demo/demo.md:227]: I don't think the `setCustomValidity` example works as intended. The only way to trigger the error message is by clicking in and out of the input without typing anything, which is misleading due to the error message. The validity works as intended and is not included in this PR, but I wanted to bring that to your attention.

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

File: src/base-input.js (hunk 1 above)
Comment (jason-capsule42) [reply src/base-input.js:711]: It means that `this.setCustomValidity` will be an empty string if `this.setCustomValidityMissing` is undefined.

Comment (blackfalcon) [reply src/base-input.js:711]: The `||` is an `or` operator. E.g. `this.setCustomValidity` will equal the value of `this.setCustomValidityValueMissing` OR an empty string.

File: src/base-input.js [hunk 4]
```diff
@@ -767,14 +767,12 @@ export default class BaseInput extends LitElement {
    * @returns {void}
    */
   configureDataForType() {
-    if (this.type === 'month-day-year') {
+    if (this.type === 'month-day-year' || 'year-month-day') {
```
Comment (blackfalcon) [inline src/base-input.js:770]: These updates broke tests 
https://github.com/AlaskaAirlines/auro-input/actions/runs/4640862496/jobs/8213233895#step:5:95

This should have been picked up in the pre-commit?

**Review by blackfalcon — DISMISSED** [review]

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

**Comment by jason-capsule42** [comment]
@jordanjones243 The reason for the ` || ''` code on the setCustomValidity had to do with when datepicker was trying to get the error message text from auro-input. The process overall worked better when input didn't have it set to undefined after executing validation.

**Review by jordanjones243 — APPROVED** [review]

**Pull request merged**